homeassistant/components/light/limitlessled.py

```diff
@@ -8,171 +8,273 @@
+    # maintain backwards compatibility.
+    legacy_setup(config, add_devices_callback)
 
-    @staticmethod
-    def factory(pool, controller_id, group, name, group_type):
-        ''' Construct a Limitless LED of the appropriate type '''
-        if group_type == 'white':
-            return WhiteLimitlessLED(pool, controller_id, group, name)
-        elif group_type == 'rgbw':
-            return RGBWLimitlessLED(pool, controller_id, group, name)
-
-    # pylint: disable=too-many-arguments
-    def __init__(self, pool, controller_id, group, name, group_type):
-        self.pool = pool
-        self.controller_id = controller_id
-        self.group = group
+    # Use the expanded configuration format.
+    if 'bridges' not in config:
+        return
+    lights = []
+    for bridge_conf in config.get('bridges'):
+        if 'groups' not in bridge_conf:
+            continue
+        bridge = Bridge(bridge_conf.get('host'),
+                        port=bridge_conf.get('port', DEFAULT_PORT),
+                        version=bridge_conf.get('version', DEFAULT_VERSION))
+        for group_conf in bridge_conf.get('groups'):
+            group = bridge.add_group(group_conf.get('number'),
+                                     group_conf.get('name'),
+                                     group_conf.get('type', DEFAULT_LED_TYPE))
+            lights.append(LimitlessLEDGroup.factory(group))
+    add_devices_callback(lights)
 
-        self.pool.execute(self.controller_id, "set_group_type", self.group,
-                          group_type)
 
-        # LimitlessLEDs don't report state, we have track it ourselves.
-        self.pool.execute(self.controller_id, "off", self.group)
+def state(new_state):
+    """ State decorator.
+
+    Specify True (turn on) or False (turn off).
+    """
+    def decorator(function):
+        """ Decorator function. """
+        # pylint: disable=no-member
+        def wrapper(self, **kwargs):
+            """ Wrap a group state change. """
+            pipeline = Pipeline()
+            transition_time = DEFAULT_TRANSITION
+            # Stop any repeating pipeline.
+            if self.repeating:
+                self.repeating = False
+                self.group.stop()
+            # Not on? Turn on.
+            if not self.is_on:
+                pipeline.on()
+            # Set transition time.
+            if ATTR_TRANSITION in kwargs:
+                transition_time = kwargs[ATTR_TRANSITION]
+            # Do group type-specific work.
+            function(self, transition_time, pipeline, **kwargs)
+            # Update state.
+            self.on_state = new_state
+            self.group.enqueue(pipeline)
+            self.update_ha_state()
+        return wrapper
+    return decorator
+
+
+class LimitlessLEDGroup(Light):
+    """ LimitessLED group. """
+    def __init__(self, group):
+        """ Initialize a group. """
+        self.group = group
+        self.repeating = False
+        self.on_state = False
```

Comment:
I think this name is a bit confusing, it sounds like a callback when state happens. Maybe just call it `self._is_on` to be in line with `_brightness`?